### PR TITLE
Implement POST /users endpoint

### PR DESCRIPTION
**Pull request by schi314**
Allow user creation through POST /users

For #92

**Testing Notes:**
Implemented integration tests and unit tests, with results shown below:

---
image
Integration test that targets HTTP request/response
* Fails 3 test cases because they require GET /users implementation to test user information are correctly stored

Changes made to allow testing:
1. Included `axios` to make HTTP request to generate auth token before running any tests
2. Added `FIREBASE_TEST_AUTH_KEY`, `FIREBASE_TEST_AUTH_EMAIL`, and `FIREBASE_TEST_AUTH_PASS` to `.env` file. These are used for generating auth token.

---
image
* Unit test that targets data storage
---
image
image
* Fails two other tests but are not considered for this PR

---

### CODE REVIEW

**Review by lanxdrey19 — APPROVED** [review]
LGTM

File: backend/src/controllers/user.controller.ts [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
+    const createdUser = await userService.createUser(req.body);
 
-   return user;
- }
\ No newline at end of file
+    // Don't return auth_id in the response
+    res.status(httpStatus.CREATED).json({
+      _id: createdUser._id,
+      first_name: createdUser.first_name,
+      last_name: createdUser.last_name,
```
Comment (butterplup) [inline backend/src/controllers/user.controller.ts:28]: are we still using a `first_name` and `last_name` field rather than simply a `full_name` field?

Comment (butterplup) [reply backend/src/controllers/user.controller.ts:28]: there seems to be some inconsistency between first/last vs full in the test cases also.

Comment (schi314) [reply backend/src/controllers/user.controller.ts:28]: User has both `first_name` and `last_name`, but Person is the one that changed to having `full_name` only.

**Review by sdiv877 — APPROVED** [review]
LGTM 👍

**Review by butterplup — APPROVED** [review]
LGTM

**Pull request merged**